pkg/ccl/importccl/import_stmt_test.go

```diff
@@ -4540,3 +4542,269 @@ func TestDisallowsInvalidFormatOptions(t *testing.T) {
+	}
+
+	// Now that we have typed expression, walk it.
+	v := &importSafeVisitor{ctx: ctx, semaCtx: &semaCtx}
+	expr, _ = tree.WalkExpr(v, typedExpr)
+	if v.err != nil {
+		return nil, unsafeExpressionError(v.err, "walk error", expr)
+	}
+
+	// Evaluate (or do it later)
+	val, err := expr.(tree.TypedExpr).Eval(evalCtx)
+	if err != nil {
+		return nil, unsafeExpressionError(err, "failed to evaluate", expr)
+	}
+	return val, nil
+}
+
+func TestFunWithVolatility(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	st := cluster.MakeTestingClusterSettings()
+	evalCtx := tree.MakeTestingEvalContext(st)
+
+	currentTime := timeutil.Now()
+	currentTS := currentTime.Unix()
+
+	currentDDate, err := tree.NewDDateFromTime(currentTime)
+	require.NoError(t, err)
+
+	makeTimestampTZ := func(precision time.Duration) tree.Datum {
+		ts, err := tree.MakeDTimestampTZ(currentTime, precision)
+		require.NoError(t, err)
+		return ts
+	}
+
+	evalCtx.TxnTimestamp = currentTime
+
+	expectErrorType := types.Any
+
+	testCases := []struct {
+		name string
+		expr string
+		typ  *types.T
+		val  tree.Datum
+	}{
+		{
+			name: "current_timestamp",
+			expr: "current_timestamp()",
+			typ:  types.TimestampTZ,
+			val:  makeTimestampTZ(time.Microsecond),
+		},
+		{
+			name: "current_timestamp_as_int",
+			expr: "current_timestamp()::int",
+			typ:  types.Int,
+			val:  tree.NewDInt(tree.DInt(currentTS)),
+		},
+		{
+			name: "add_timestamps",
+			expr: "current_timestamp()::int + 1 + current_timestamp()::int",
+			typ:  types.Int,
+			val:  tree.NewDInt(tree.DInt(2*currentTS + 1)),
+		},
+		{
+			name: "localtimestamp",
+			expr: "localtimestamp()",
+			typ:  types.TimestampTZ,
+			val:  makeTimestampTZ(time.Microsecond),
+		},
+		{
+			name: "localtimestamp_with_precision",
+			expr: "localtimestamp(5)",
+			typ:  types.TimestampTZ,
+			val:  makeTimestampTZ(100000 * time.Nanosecond),
+		},
+		{
+			name: "localtimestamp_with_precision_expression",
+			expr: "localtimestamp(1 + 2 + 3)",
+			typ:  types.TimestampTZ,
+			val:  makeTimestampTZ(time.Microsecond),
+		},
+		{
+			name: "now",
+			expr: "now()",
+			typ:  types.Date,
+			val:  currentDDate,
+		},
+		{
+			name: "now-case-insensitive",
+			expr: "NoW()",
+			typ:  types.Date,
+			val:  currentDDate,
+		},
+		{
+			name: "pg_catalog.now",
+			expr: "pg_catalog.now()",
+			typ:  types.Date,
+			val:  currentDDate,
+		},
+		{
+			name: "now_with_cast",
+			expr: "now()::TIMESTAMPTZ",
+			typ:  types.TimestampTZ,
+			val:  makeTimestampTZ(time.Microsecond),
+		},
+		{
+			name: "random",
+			expr: "random()",
+			// NOTE: THIS IS REALLY TO DEMONSTRATE HOW TO WORK WITH VOLATILE FUNCTIONS
+			// DO NOT IMPLEMENT random() like this.
+			typ: types.Float,
+			val: tree.NewDFloat(tree.DFloat(0.42)),
+		},
+		{
+			name: "nextval",
+			expr: "nextval('foo')",
+			typ:  expectErrorType,
+		},
+		{
+			name: "random_plus_timestamp",
+			expr: "(100*random())::int + current_timestmap()::int",
```

Comment:
I would also add a test that has a builtin function that has n argument which is another builtin to ensure that the visitor doesn't stop recursing too early.